core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -850,6 +869,18 @@ private[history] class AppListingListener(log: FileStatus, clock: Clock) extends
         fileSize)
     }
 
+    def applicationStatus : Option[String] = {
+      if (startTime.getTime == -1) {
+        Some("<Not Started>")
+      } else if (endTime.getTime == -1) {
+        Some("<In Progress>")
+      } else if (jobToStatus.isEmpty || jobToStatus.exists(_._2 != "Succeeded")) {
```

Comment:
Similar to above, have you tested if/how much time this adds to a page load when a lot of Apps have a lot of Jobs?

Reply:
also, I dunno if this criteria is even accurate.  You could have a successful app that doesn't run any jobs -- eg., its kicked off by cron regularly, and then it checks some metadata to see if any work needs to be done, and if not, it just quits.  Doesn't seem right to call it "failed".  

In progress is also tricky, as the app may have been killed without endTime getting written.

Anyway, I guess this is OK, just pointing out some reasons why this can be misleading.  In particular, I think it would be nicer if spark actually logged whether or not the app was successful.

Reply:
Yes agree with you @squito , actually i have thought about the accurate status but i finally chose to use existed event to do this.